cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -321,46 +363,94 @@ impl Cmd {
         fs::write(target_file_path, wasm_bytes).map_err(Error::WritingWasmFile)
     }
 
-    fn print_build_summary(print: &Print, target_file_path: &PathBuf) -> Result<(), Error> {
-        print.infoln("Build Summary:");
-        let rel_target_file_path = target_file_path
+    fn print_build_summary(
+        print: &Print,
+        target_file_path: &PathBuf,
+        optimized_path: Option<&PathBuf>,
+    ) -> Result<(), Error> {
+        if optimized_path.is_some() {
+            print.infoln("Build Summary (Optimized):");
+        } else {
+            print.infoln("Build Summary:");
+        }
```

Comment:
I think we can keep the output consistent for the top half.
```suggestion
        print.infoln("Build Summary:");
```